tests/h2_test_helper.c

```diff
@@ -458,7 +458,7 @@ static struct aws_h2_decoder_vtable s_decoder_vtable = {
     .on_push_promise_begin = s_decoder_on_push_promise_begin,
     .on_push_promise_i = s_decoder_on_push_promise_i,
     .on_push_promise_end = s_decoder_on_push_promise_end,
-    .on_data = s_decoder_on_data,
+    .on_data_i = s_decoder_on_data_i,
```

Comment:
Could you please modify the decode_tester to do testing on this API change? In addition to storing info about the callbacks that happen, this code also validates that the callbacks happen in the order they're supposed to.

Check out how we do things with the other _begin()/i()/end() callbacks work

Instead of creating a new `h2_decoded_frame` per on_data_i() call, create a new frame in on_data_begin() and append to its data as on_data_i() callbacks happen. End the frame when on_data_end() is called.

Reply:
Oh, I just realized it after I made the PR. And it's actually already there. Sorry for the confusion.